### ISSUE TITLE
[Conduits] PATCH conduit does not properly "delete" shards

### ISSUE THREAD

**Issue by Marenthyu** [twitchdev/issues#914]
**Brief description**
Under certain circumstances, Conduits do not seem to "scale down" as described in the documentation, keeping shards above the maximum shard count "alive" instead.

**How to reproduce**
1. Create a Conduit with 2 Shards
![](https://lowee.de/Postman_2024-02-12_20-09-52.png)
2. Connect 2 WebSocket Transports to the Shards
![](https://lowee.de/Postman_2024-02-12_20-15-28.png)
3. Disconnect WebSocket with Shard ID 0 (unsure at the time of writing if required)
![](https://lowee.de/chrome_2024-02-12_20-36-04.png)
4. Scale down Conduit to 1 Shard
![](https://lowee.de/Postman_2024-02-12_20-36-27.png)
5. Observe unchanged Conduit Status
![](https://lowee.de/2024-02-12_20-36-56.png)

**Expected behavior**
Shard with ID 1 should have been disabled/deleted and either a "status" of "deleted", "scaled_down" or similar should be set or not be present in the list of shards anymore.

**Additional context or questions**
Refer to [Discussion on Discord in #eventsub-help](https://discord.com/channels/504015559252377601/776875921654546463/1206354674727391344)

**Comment by BarryCarlyon** [comment]
To add a smidge of clarity on this one and a repo case:

1. Create a conduit
2. Subscribe some stuff to the conduit (chat probably works best as "noisy")
4. Give the conduit two shards
5. Assign those shards
6. Wait for data and confirm both shards getting data
7. Scale the conduit down to 1 shard
8. All data is moved to the first shard, so the second shard is "effectivly disabled"
9. Shard 2 still shows in the [Get Conduit Shards](https://dev.twitch.tv/docs/api/reference/#get-conduit-shards) API but is still "enabled" however is not being sent data (which is good, no "leaks" per sae)
10. In this case shard 2 is a websocket, I kill the websocket and the shard becomes `websocket_disconnected` in the shards response

So, in summary, when you scale down using "reduce shard count"

- The "extra" shards stop receiving data.
- The "extra" shards still appear in [Get Conduit Shards](https://dev.twitch.tv/docs/api/reference/#get-conduit-shards)
- The "extra" shards will still "change status" if the transport assigned changes state (which conceptually will be never for a webhook, but more likely on a websocket)

**Comment by BlueCakeWithCows** [comment]
This should now be fixed, though existing orphaned shards may still exist.
